.github/actions/setup-linux/action.yml

```diff
@@ -47,6 +47,39 @@ runs:
             echo "Starting docker daemon..." && sudo systemctl start docker;
         fi
 
+    - name: Install uv
+      uses: astral-sh/setup-uv@374b9305c53f17cee51e793c1234cdb1897489d5 # v7
+      if: ${{ steps.check_container_runner.outputs.IN_CONTAINER_RUNNER == 'false' }}
+      with:
+        python-version: 3.12
+
+    - name: Install awscli
+      shell: bash
+      if: ${{ steps.check_container_runner.outputs.IN_CONTAINER_RUNNER == 'false' }}
+      run: |
+        set -euo pipefail
+        if command -v aws &> /dev/null; then
+          echo "awscli is already installed."
+        else
+          echo "awscli not found, installing..."
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          uv run python -c "import zipfile; zipfile.ZipFile('awscliv2.zip').extractall()"
+          sudo ./aws/install
+          rm -rf awscliv2.zip aws
+        fi
+
+    - name: Install pip
+      shell: bash
+      if: ${{ steps.check_container_runner.outputs.IN_CONTAINER_RUNNER == 'false' }}
+      run: |
+        set -euo pipefail
+        if command -v pip3 &> /dev/null; then
+          echo "pip3 is already installed."
+        else
+          echo "pip3 not found, installing..."
+          uv run pip install pip==25.3 --system
+        fi
```

Comment:
This sounds AI hallucinated, why `pip` would be absent from `install uv` step?

Reply:
`uv pip` is installed, but this doesn't mean that `pip` is installed.